RedHatInsights/vulnerability-ui
# VULN-2391: OS filter displays only relevant values obtained from API (1/2)

Implements https://issues.redhat.com/browse/VULN-2391.

- [x] /systems page: make OS filter values dynamic
- [x] /vulnerability/cves/{cve_id}: make OS filter values dynamic
- [x] Add tests to SystemsExposedTableToolbar

## How to test:

1. Navigate to https://stage.foo.redhat.com:1337/beta/insights/vulnerability/systems.
2. Verify GET  /system_profile/operating_system is made.
3. Check whether OS filter matches the API response.
4. Check basic filter functions (chips loaded, values selected, table updated, URL search params updated, filters can be removed).
5. Navigate to any of CVEs page.
6. Do the same steps as for the systems list page.

## Screenshots

![image](https://user-images.githubusercontent.com/31385370/189340056-dce64d8c-31b1-4154-998b-7f0c2c9dd6d8.png)
![image](https://user-images.githubusercontent.com/31385370/189339983-d158a78b-4b1f-4c8a-8534-7dff1aefc2ff.png)
